### PR TITLE
Add a script for validating collated single-html files

### PR DESCRIPTION
**Pull request by pumazi**

### CODE REVIEW

**Comment by pumazi** [comment]
I'll look into the failing tests... 😸

File: cnxepub/scripts/collated_single_html/main.py [hunk 1]
```diff
@@ -5,31 +5,51 @@
...
+
     args = parser.parse_args(argv)
 
-    binder = cnxepub.adapt_single_html(args.collated_html.read())
-    pprint(cnxepub.model_to_tree(binder))
+    html = args.collated_html.read()
+    try:
+        htree = etree.fromstring(html)
+    except etree.XMLSyntaxError:
+        htree = etree.HTML(html)
```
Comment (karenc) [inline cnxepub/scripts/collated_single_html/main.py]: Should this be inside `cnxepub.adapt_single_html`?

Comment (karenc) [inline cnxepub/scripts/collated_single_html/main.py]: And should we use `cnxepub.reconstitute` instead of `cnxepub.adapt_single_html`? :D

Comment (pumazi) [inline cnxepub/scripts/collated_single_html/main.py]: Is that a suggestion or question? I'd think you would know best, seeing as you wrote the `cnxepub.adapt_single_html` function.

Comment (pumazi) [inline cnxepub/scripts/collated_single_html/main.py]: Ah, now that's an excellent idea. Perhaps adapt_single_html should remain xhtml specific at this time. Yeah, let's use `cnxepub.collation.reconstitute` and push this try..except there.

I'll take care of that change.

Comment (karenc) [inline cnxepub/scripts/collated_single_html/main.py]: IMO there's no point in doing this in the validator script and then failing when we actually reconstitute.

**Comment by pumazi** [comment]
⭐ Good work @karenc. Feel free to rebase it.

@reedstrm does this still work for you? If so, @karenc can move forward with the merge.

**Comment by reedstrm** [comment]
I'll get Helene to test it. They've been using branch-as-of-10am successfully all day. :-)

File: cnxepub/scripts/collated_single_html/main.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+"""Processes a collated book's single-html representation for validation.
+
+Ensures safe parsing for specific elements.
+"""
+from __future__ import print_function
+import argparse
+import logging
+import sys
+from pprint import pformat
+from lxml import etree
```
Comment (reedstrm) [inline cnxepub/scripts/collated_single_html/main.py]: unused tree import, I think?

**Comment by reedstrm** [comment]
Works on my test case cooked html.  Only niggle is the now unused etree import.

**Comment by reedstrm** [comment]
I'm willing to merge, since all my code's been refactored away ... :-)

**Pull request merged**